testutil/ciao-down/download.go

```diff
@@ -0,0 +1,139 @@
+/*
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+)
+
+// Constants for the Guest image used by ciao-down
+
+const (
+	guestDownloadURL       = "https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img"
+	guestImageName         = "xenial-server-cloudimg-amd64-disk1.img"
+	guestImageTmpName      = "xenial-server-cloudimg-amd64-disk1.img.part"
+	guestImageFriendlyName = "Ubuntu 16.04"
+)
+
+type progressCB func(p progress)
+
+type progress struct {
+	downloadedMB int
+	totalMB      int
+}
+
+type progressReader struct {
+	downloaded int64
+	totalMB    int
+	reader     io.Reader
+	cb         progressCB
+}
+
+func (pr *progressReader) Read(p []byte) (int, error) {
+	read, err := pr.reader.Read(p)
+	if err == nil {
+		oldMB := pr.downloaded / 10000000
+		pr.downloaded += int64(read)
+		newMB := pr.downloaded / 10000000
+		if newMB > oldMB {
+			pr.cb(progress{downloadedMB: int(newMB * 10), totalMB: pr.totalMB})
+		}
+	}
+	return read, err
+}
+
+func getUbuntu(ctx context.Context, dest io.WriteCloser, cb progressCB) (err error) {
+	defer func() {
+		err1 := dest.Close()
+		if err == nil && err1 != nil {
+			err = err1
+		}
+	}()
+	req, err := http.NewRequest("GET", guestDownloadURL, nil)
```

Comment:
@markdryan this fails on clear for some reason with error 

`Unable download file https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img: Get https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img: x509: certificate signed by unknown authority
`

curl to the same URL on the host works fine.